src/actions/guides/database/querying.cr

```diff
@@ -7,294 +7,428 @@ class Guides::Database::Querying < GuideAction
+
+    For our examples, we will use this `User` model.
+
+    ```crystal
+    class User < BaseModel
+      table :users do
+        # `id`, `created_at`, `updated_at` predefined for us
+        column name : String
+        column age : Int32
+        column admin : Bool
+      end
+    end
+    ```
+
+    ### Select shortcuts
 
-    * `first` - will raise if a record is not found
-    * `first?` - for when you want to allow `nil` to be returned
+    By default, all query objects include `Enumerable(T)` methods like `each`, and `map`.
+    All query methods are called on the instance of the query object, but there's also a
+    few class methods for doing quick finds.
+
+    * `first` - Returns the first record. Raise `Avram::RecordNotFoundError` if no record is found.
+    * `first?` - Returns the first record. Returns `nil` if no record is found.
+    * `find(id)` - Returns the record with the primary key `id`. Raise `Avram::RecordNotFoundError` if no record is found.
+    * `last` - Returns the last record. Raise `Avram::RecordNotFoundError` if no record is found.
+    * `last?` - Returns the last record. Returns `nil` if no record is found.
+    * `all` - Returns an array of all the records.
+
+    ```crystal
+    first_user = UserQuery.first
+    last_user = UserQuery.last
+    specific_user = UserQuery.find(4)
+    all_users = UserQuery.all
+    ```
+
+    ### Lazy loading
+
+    The query does not actually hit the database until a method is called to fetch a result
+    or iterate over results.
+
+    The most common methods are:
+
+    * `first`
     * `find`
-    * All methods included by `Enumerable(T)`, such as `each` and `map`
+    * `each`
 
-    It also has methods for ordering and adding where clauses. These are generated
-    based on the columns defined in your model.
+    For example:
 
-    For the model we defined Lucky would generate:
+    ```crystal
+    # The query is not yet run
+    query = UserQuery.new
+    query.name("Sally")
+    query.age(30)
 
-    * `created_at` (automatically added)
-    * `updated_at` (automatically added)
-    * `id` (automatically added)
-    * `age`
-    * `name`
+    # The query will run once `each` is called
+    # Results are not cached so a request will be made every time you call `each`
+    query.each do |user|
+      pp user.name
+    end
+    ```
+
+    ## Simple Selects
+
+    > When doing a `SELECT`, Avram will select all of the columns individually (i.e. `users.id,
+    > users.created_at, users.updated_at, users.name, users.age, users.admin`) as opposed to `*`.
+    > However, for brevity, we will use `COLUMNS`.
 
+    ### Select all
 
-    ## Distinct on
+    `SELECT COLUMNS FROM users`
 
     ```crystal
-    UserQuery.new.distinct_on(&.name)
+    users = UserQuery.new.all
     ```
 
-    ## Where clauses
+    ### Select first
 
-    You can use the methods generated by Lucky to create where clauses.
+    `SELECT COLUMNS FROM users LIMIT 1`
```

Comment:
I don't think so https://github.com/luckyframework/avram/blob/master/src/avram/queryable.cr#L96-L97

It just adds a `limit(1)`, and then calls the `.first?` method from Array. Maybe it should?

Reply:
Mmm I see what's happening. I was confused as to why `first` didn't have ordering but `last` did. I think it's because `last` adds `reverse_order` which sets an order by clause. So, not a problem here, but I'll put an issue in on Avram.

Reply:
https://github.com/luckyframework/avram/issues/115